pipeline/inferrer/inference_manager/src/main/scala/uk/ac/wellcome/platform/inference_manager/services/InferrerAdapter.scala

```diff
@@ -0,0 +1,90 @@
+package uk.ac.wellcome.platform.inference_manager.services
+
+import akka.http.scaladsl.model._
+import akka.http.scaladsl.unmarshalling.Unmarshal
+import akka.stream.Materializer
+import de.heikoseeberger.akkahttpcirce.FailFastCirceSupport._
+import grizzled.slf4j.Logging
+import io.circe.Decoder
+import io.circe.generic.semiauto.deriveDecoder
+import uk.ac.wellcome.models.work.internal.{
+  AugmentedImage,
+  InferredData,
+  MergedImage,
+  Minted
+}
+import uk.ac.wellcome.platform.inference_manager.models.FeatureVectorInferrerResponse
+
+import scala.concurrent.Future
+
+/*
+ * An InferrerAdapter is specific to the inferrer and the data that is being augmented.
+ * Implementors must provide:
```

Comment:
Is a model where the inferrer request/response is standardised feasible (and desirable)?

Reply:
We did discuss some kind of standardised response in which the inferrer also instructs the manager where to put the data in the work/image, or that the data always goes in a fixed place, but we decided that the inferrer shouldn't know what works/images look like.

fwiw that's also why the manager does the work of manipulating the data from the inferrer (in this case, splitting the feature vector), as we also didn't want the inferrer to know about (for example) ElasticSearch's dense vector limitations.

Another thing is that it would also be nice to have other, and potentially open, deployments of the inferrer somewhere, in which case a descriptive response format is probably preferable.

Reply:
Basically I agree it could be nice but marginally fall on the side of thinking it isn't desirable, up for debate though

Reply:
Interested in a high level opinion - not an issue blocking this PR.

Reply:
> Basically I agree it could be nice but marginally fall on the side of thinking it isn't desirable, up for debate though

👍 Probably trying for a standard request/response now would be over-engineering the solution - as we add more inferrers it'll focus the problem.